src/DoctrineBundle.php

```diff
@@ -168,4 +170,15 @@ public function shutdown()
     public function registerCommands(Application $application)
     {
     }
+
+    public function getPath(): string
+    {
+        /** @psalm-suppress RedundantPropertyInitializationCheck */
+        if (! isset($this->path)) {
+            $reflected  = new ReflectionObject($this);
+            $this->path = dirname($reflected->getFileName(), 2);
```

Comment:
we could simplify the implementation a lot with `return dirname(__DIR__)` here instead of using Reflection to find the source of the class.

Reply:
not a fan of `dirname()`, since it's a function call right? Why not just `__DIR__.'/..'`?

And let's do one better and remove caching like so
```php
public function getPath(): string
{
  return `__DIR__.'/..';
}
```

since caching such a simple logic to `$path` with no method calls is pointless

Reply:
@ostrolucky `dirname` with a known path (like `__DIR__`) gets computed at compile-time by OPCache, so no overhead.

`__DIR__.'/..'` will not be the absolute path, which might cause issues if some places are checking for paths starting with this path.

Reply:
I see. Thanks for explanation. Sorry @greg0ire !